tools/src/rpc-schema/openrpc/mod.rs

```diff
@@ -0,0 +1,103 @@
+use std::collections::BTreeMap;
+
+use serde_json::{Map, Value};
+
+use self::document::{
+    Components, ContactObject, InfoObject, LicenseObject, MethodObject, Openrpc, OpenrpcDocument,
+};
+use crate::parser::{ParsedItemStruct, ParsedTraitItemFn};
+
+#[allow(unused)]
+pub mod document;
+
+/// A builder for constructing an OpenRPC document.
+#[derive(Clone)]
+pub struct OpenRpcBuilder {
+    open_rpc_doc: OpenrpcDocument,
+    structs: Vec<ParsedItemStruct>,
+    methods: Vec<ParsedTraitItemFn>,
+}
+
+impl OpenRpcBuilder {
+    /// Adds a schema based on a Rust struct to the OpenRPC document.
+    pub fn with_schema(mut self, item_struct: &ParsedItemStruct) -> OpenRpcBuilder {
+        self.structs.push(item_struct.clone());
+        self
+    }
+
+    /// Adds a method based on a Rust trait method to the OpenRPC document.
+    pub fn with_method(mut self, item_fn: &ParsedTraitItemFn) -> OpenRpcBuilder {
+        self.methods.push(item_fn.clone());
+        self
+    }
+
+    /// Sets the version of the OpenRPC document.
+    pub fn version(mut self, version: String) -> OpenRpcBuilder {
+        self.open_rpc_doc.info.version = version;
+        self
+    }
+
+    /// Creates a new instance of OpenRpcBuilder with default settings and returns it.
+    pub fn builder() -> OpenRpcBuilder {
+        OpenRpcBuilder{ open_rpc_doc: OpenrpcDocument {
+            openrpc: Openrpc::V26,
+            info: InfoObject {
+                title: "Nimiq JSON-RPC Specification".to_string(),
+                description: Some("Through the use of JSON-RPC, Nimiq nodes expose a set of standardized methods and endpoints that allow external applications and tools to interact, stream and control the behavior of the nodes. This includes functionalities such as retrieving information about the blockchain state, submitting transactions, managing accounts, and configuring node settings.".to_string()),
+                version: "Testnet 0.20.0".to_string(),
+                terms_of_service: None,
+                contact: Some(ContactObject { name: Some("The Nimiq Foundation".to_string()), email: Some("info@nimiq.com".to_string()), url: Some("https://nimiq.com".to_string()) }),
```

Comment:
Same as contact and license. Wondering if we can get them from`Cargo.toml`

Reply:
Contact and licence are now extracted from the `Cargo.toml`.